### PR TITLE
feat: add last updated timestamp to mapping ui

### LINKED ISSUE #1034: Add last updated timestamp to `Mappings`
Show the last updated time for mappings that were updated (+updated by?)

### PR DESCRIPTION
**Pull request by pehlicd**
Closes #1034 

## ✅ Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep/api/models/db/mapping.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ class MappingRule(SQLModel, table=True):
         sa_column=Column(JSON),
         nullable=False,
     )  # max_length=204800)
+    updated_by: Optional[str] = Field(max_length=255, default=None)
+    last_updated_at: datetime = Field(default_factory=datetime.utcnow)
```
Comment (talboren) [inline keep/api/models/db/mapping.py:29]: We need a migration for that 😭

File: keep-ui/app/mapping/rules-table.tsx [hunk 2]
```diff
@@ -95,6 +95,11 @@ export default function RulesTable({ mappings, editCallback }: Props) {
         </div>
       ),
     }),
+    columnHelper.display({
+      id: "updated at",
+      header: "Updated At",
+      cell: (context) => context.row.original.last_updated_at,
+    }),
```
Comment (talboren) [inline keep-ui/app/mapping/rules-table.tsx:102]: Perhaps, if you want, you can check out what I did in Extractions (where you can "open" the row and see this details there)

File: keep/api/routes/mapping.py [hunk 3]
```diff
@@ -108,6 +111,8 @@ def update_rule(
     existing_rule.matchers = rule.matchers
     existing_rule.file_name = rule.file_name
     existing_rule.priority = rule.priority
+    existing_rule.updated_by = authenticated_entity.email
+    existing_rule.last_updated_at = datetime.datetime.utcnow()
```
Comment (talboren) [inline keep/api/routes/mapping.py:115]: https://news.ycombinator.com/item?id=38333116
```suggestion
    existing_rule.last_updated_at = datetime.datetime.now(tz=datetime.timezone.utc)
```

**Review by talboren — CHANGES_REQUESTED** [review]
small few comments ❤️

File: keep/api/models/db/mapping.py (hunk 1 above)
Comment (pehlicd) [reply keep/api/models/db/mapping.py:29]: Added migration

File: keep/api/routes/mapping.py (hunk 3 above)
Comment (pehlicd) [reply keep/api/routes/mapping.py]: Wow didn't know that it is deprecated, fixed it.

File: keep-ui/app/mapping/rules-table.tsx (hunk 2 above)
Comment (talboren) [reply keep-ui/app/mapping/rules-table.tsx:102]: @pehlicd wdyt about this one? if you chose not to, maybe also add edited by?

Comment (pehlicd) [reply keep-ui/app/mapping/rules-table.tsx:102]: @talboren I just don't know what to check about it bro, could you share more insights about it?

Comment (pehlicd) [reply keep-ui/app/mapping/rules-table.tsx]: I updated it like extraction table fyi @talboren

File: keep/api/routes/mapping.py [hunk 4]
```diff
@@ -57,6 +58,8 @@ def create_rule(
         **rule.dict(),
         tenant_id=authenticated_entity.tenant_id,
         created_by=authenticated_entity.email,
+        updated_by=authenticated_entity.email,
+        last_updated_at=datetime.datetime.utcnow(),
```
Comment (talboren) [inline keep/api/routes/mapping.py:62]:
```suggestion
```

**Pull request merged**